frontend/plan/src/syncutils.js

```diff
@@ -0,0 +1,101 @@
+import {
+    attemptDeletion,
+    createScheduleOnBackend,
+    deleteSchedule,
+    deleteScheduleOnBackend,
+    deletionAttemptCompleted,
+    deletionSuccessful,
+    fetchSchedulesAndInitializeCart,
+    incrementDeletionAttempts,
+    updateScheduleOnBackend
+} from "./actions";
+import fetch from "cross-fetch";
+import getCsrf from "./csrf";
+
+// Ensure that fetches only happen once ever 250 ms
+let lastFetched = 0;
```

Comment:
I think the design of this function could be improved - instead of returning null when theres a delay, consider having conditional fetch take in a function as a parameter and call the function only if the condition is met, where the function will call the fetch with the promise chain.

Reply:
This function is only being used for fetches though. The extra complexity required to do what you're describing isn't worth it in my opinion

Reply:
nvm you're right, that would be cleaner, now that I think about it more